client/src/main/java/com/netflix/conductor/client/automator/PollingSemaphore.java

```diff
@@ -42,20 +42,33 @@ boolean canPoll() {
         return acquired;
     }
 
-    /** Signals that processing is complete and the permit can be released. */
-    void complete() {
+    /** Signals that processing is complete and the specified number of permits can be released. */
+    void complete(int numSlots) {
         LOGGER.debug("Completed execution; releasing permit");
-        semaphore.release();
+        semaphore.release(numSlots);
     }
 
     /**
      * Gets the number of threads available for processing.
      *
      * @return number of available permits
      */
-    int availableThreads() {
+    int availableSlots() {
         int available = semaphore.availablePermits();
         LOGGER.debug("Number of available permits: {}", available);
         return available;
     }
+
+    /**
+     * Signals if processing is allowed based on whether specified number of permits can be
+     * acquired.
+     *
+     * @param numSlots the number of permits to acquire
+     * @return {@code true} - if permit is acquired {@code false} - if permit could not be acquired
+     */
+    public boolean acquireSlots(int numSlots) {
```

Comment:
Is it possible to delete the `canPoll()` method above now that this method is being used instead?

Reply:
ya, removed now